auto_nag/nag_me.py

```diff
@@ -199,6 +199,12 @@ def prepare_mails(self):
                 bug_data = info[person]
                 data += bug_data
 
+            # The same bug can be several times in the list
+            # because we send an email to a team.
+            # So remove dups in tuplifying them (to have hashable types)
+            data = {tuple(d.items()) for d in data}
+            data = [dict(d) for d in data]
```

Comment:
Couldn't you avoid to have duplicates in the first place by having a set of bug IDs and not adding the bug to data if the bug ID is already in the set?

Reply:
The goal of this class is to gather people data with for example the same manager.
So each person may have different bugs and so we've a dict person => [...bug_data]
So each time I add a bug_data for one person it means that I'd need to iterate on this dict and list to find if this bug_data is there... I think it's easier to remove dups at the end.
Maybe we could refactor a little to avoid such a situation...
Anyway this bug appeared because we send mail to all the triagers (introduced by patch to handle layout team) for step 2 in no_priority.

Reply:
Couldn't you do something like:
```
        for manager, info in self.data.items():
            added_bug_ids = set()
            data = []
            To = sorted(info.keys())
            for person in To:
                data += (bug_data for bug_data in info[person] if bug_data['id'] not in added_bug_ids)
                added_bug_ids.update(bug_data['id'] for bug_data in info[person])
```